.github/actions/comment/action.yml

```diff
@@ -58,23 +74,60 @@ runs:
       # We use the +H option to prevent history expansion, so the '!' character can be printed out without 
       # problems.
       run: |
+        # Parse URL
+        if [[ -n "${{ inputs.url }}" ]]; then
+          url= xargs <<< "${{ inputs.url }}" # Remove leading/trailing whitespaces
+          # Regex for the url in an expanded form
+          regex='^'                             # Beginning of string anchor
+          regex+='https://github\.com/'         # Github URL
+          regex+='([^/ ]+)'                     # Match any non-space/non-slash characters - Capture group 1 (the OWNER)
+          regex+='/'                            # Match a forward slash
+          regex+='([^/ ]+)'                     # Match any non-space/non-slash characters - Capture group 2 (the REPO)
+          regex+='/'                            # Match a forward slash
+          regex+='(issues|pull)/'               # Match "issues/" or "pull/" - Capture group 3 (not used)
+          regex+='([0-9]+)'                     # Match one or more digits - Capture group 4 (the NUMBER)
+          regex+='(#issuecomment-([0-9]+))?'    # Optionally match "#issuecomment-" and one or more digits - Capture groups 5 (not used) and 6 (the COMMENT-ID)
+          regex+='$'                            # End of string anchor
+          if [[ "$1" =~ $regex ]]; then
+            OWNER=${BASH_REMATCH[1]}
+            REPOSITORY=${BASH_REMATCH[2]}
+            NUM=${BASH_REMATCH[4]}
+            COMMENT_ID=${BASH_REMATCH[6]}
+          else
+            echo "::error::Invalid URL \"${{ inputs.url }}\". Please provide a valid URL format."
+            exit 1
+          fi
+        fi
+
         # Set NUMBER
         if [[ ${{ inputs.number }} == 0 ]]; then # If number is not provided (number type defaults to 0 -- https://docs.github.com/en/enterprise-cloud@latest/actions/writing-workflows/workflow-syntax-for-github-actions#onworkflow_callinputs)
-          if [[ -n "${{ github.event.issue }}" ]]; then
+          if [[ -n "$NUMBER" ]]; then # If url is provided
+            echo "NUMBER=$NUMBER" >> $GITHUB_ENV
+          if [[ -n "${{ github.event.issue }}" ]]; then # If in an issue event
             echo "NUMBER=${{ github.event.issue.number }}" >> $GITHUB_ENV
-          elif [[ -n "${{ github.event.pull_request }}" ]]; then
+          elif [[ -n "${{ github.event.pull_request }}" ]]; then # If in a pull_request event
             echo "NUMBER=${{ github.event.pull_request.number }}" >> $GITHUB_ENV
           else
             echo "::error::Couldn't infer PR/Issue number from the event payload, please enter PR/Issue number as an input."
             exit 1
           fi
+        elif [[ -n "${NUMBER}" ]]; then # If both number and url are provided
```

Comment:
Or better yet, short circuit much earlier in the workflow with the above, rather than check a similar conditional a bunch of times.

Reply:
Yes I meant the output of the PARSE URL. 
Actually, for consistency, I am going to rename that `NUM` to `NUMBER`.

Anyway, your suggestion for the check makes sense. 
Better to be explicit about it right at the beginning.

Reply:
Actually not:

The `-n "${{ inputs.number }}"` would not work because inputs.number is set to `0` as a default, and `[ -n "0" ]` is true.

I will check it with: `[[ "${{ inputs.number }}" != 0 && -n "${{ inputs.url }}" ]]`

Reply:
Fixed in [7fdce6e417c5e70fb3c3a0516621d448174f765b](https://github.com/ACCESS-NRI/actions/pull/14/commits/7fdce6e417c5e70fb3c3a0516621d448174f765b)